### ISSUE TITLE
Unwanted white-space in html source

### ISSUE THREAD

**Issue by jpilon** [amazonwebservices/aws-sdk-for-php#54]
For some reason this sdk is creating whitespace in the html source.

This is how my html document appears currently, the doctype should not have any leading white space.
                    <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>

I've tried finding where this happening but am unable to locate the issue. 

Also, if I copy that whitespace and paste it in a word document, it actually draws a 2 cell tablet, so I'm not sure if perhaps the issue is bad text encoding.

Anyone experience a similar issue?

**Comment by rquadling** [comment]
Sounds like a BOM.

On 10 March 2013 11:07, jpilon notifications@github.com wrote:

> For some reason this sdk is creating whitespace in the html source.
> 
> This is how my html document appears currently, the doctype should not
> have any leading white space.
> <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "
> http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
> 
> I've tried finding where this happening but am unable to locate the issue.
> 
> Also, if I copy that whitespace and paste it in a word document, it
> actually draws a 2 cell tablet, so I'm not sure if perhaps the issue is bad
> text encoding.
> 
> Anyone experience a similar issue?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/amazonwebservices/aws-sdk-for-php/issues/54
> .

## 

Richard Quadling
Twitter : @RQuadling
EE : http://e-e.com/M_248814.html
Zend : http://bit.ly/9O8vFY

**Comment by jpilon** [comment]
what does BOM stand for?

**Comment by rquadling** [comment]
On 10 March 2013 13:28, jpilon notifications@github.com wrote:

> what does BOM stand for?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/amazonwebservices/aws-sdk-for-php/issues/54#issuecomment-14681326
> .

Byte Order Mark. It is a set of bytes at the beginning of a plain text file
to indicate the byte order for UTF encoded data (Big/Little Endian stuff).
Traditionally, only for UTF-16/32/64 and not for UTF-8.

## 

Richard Quadling
Twitter : @RQuadling
EE : http://e-e.com/M_248814.html
Zend : http://bit.ly/9O8vFY

**Comment by jeremeamia** [comment]
Well... how did that creep in? I will look into this as soon as possible.

**Comment by rquadling** [comment]
Some people's editors automatically put them in. Not disastrous but
annoying as there is no visual element for the BOM byte sequence.

On 10 March 2013 17:07, Jeremy Lindblom notifications@github.com wrote:

> Well... how did that creep in. I will look into this as soon as possible.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/amazonwebservices/aws-sdk-for-php/issues/54#issuecomment-14684701
> .

## 

Richard Quadling
Twitter : @RQuadling
EE : http://e-e.com/M_248814.html
Zend : http://bit.ly/9O8vFY

**Comment by jeremeamia** [comment]
I scanned the SDK repository and I cannot find any BOMs or any other indicator of extra whitespace. This is also the only report of something like this in the SDK. Please make sure to search your project files and any other dependencies to see if something else may be at fault. Please reopen this if you identify an issue with the SDK. Thanks.

**Comment by jpilon** [comment]
After literally putting echo '1"; echo '2'; in a million files trying to trace where the white space was coming from I finally found it. In one of the files that included the aws sdk the white space was located at the closing ?> on the last line. 

The reason why I initially assumed it was the aws sdk was once I removed the include, the white space disappeared, but little did I realize that there was an sdk class reference in the same file which of course meant the rest of the file didn't load hence no white space (errors are suppressed).

I apologize for wasting everyone's time on this issue!
